### PR TITLE
Falcon UI (docs)

### LINKED ISSUE #86: falcon-ui documentation
Current work can be tracked on https://github.com/deity-io/falcon/tree/falcon-ui-docs - that includes cleanup and small improvements to ui and theme editor codebases as well.

### PR DESCRIPTION
**Pull request by tadeuszwojcik**
This pull request introduces:

- proper documentation for `@deity/falcon-ui` package
- cleanup, refactoring and couple improvements to `@deity/falcon-theme-editor` (download theme, component finder, ui related updates etc)
- updates & small fixes to  `@deity/falcon-ui`  - simplifying theme object (8px based grid by default, revert primary and secondary colors, simplyfying box shadows and border radius values) and  updating dependencies (emotion latest version) 
- updates to  `@deity/falcon-ecommerce-uikit` based on changes in `@deity/falcon-ui`

Preview of docs: https://falcon-ui-docs-vnext.netlify.com/
(closes #86 )

### CODE REVIEW

File: packages/falcon-ui/src/components/Table.mdx [hunk 1]
```diff
@@ -4,11 +4,19 @@ menu: Components
...
 import { Playground } from 'docz';
-import { Table, Thead, Tr, Th, Td, Tbody } from '../';
+import { ThemePreview } from 'docs';
+import { Table, Thead, Tr, Th, Td, Tbody } from '@deity/falcon-ui';
 
-# Table
+<ThemePreview of={[Table, Thead, Tr, Th, Td, Tbody]} />
+
+
+`Table` with it's related components (`Thead, Tr, Th, Td, Tbody`) can be used to render html `<table>` element with basic styles applied
```
Comment (Borales) [inline packages/falcon-ui/src/components/Table.mdx]:
```suggestion
`Table` with its related components (`Thead, Tr, Th, Td, Tbody`) can be used to render html `<table>` element with basic styles applied
```

File: packages/falcon-ui/src/components/Text.mdx [hunk 2]
```diff
@@ -4,11 +4,17 @@ menu: Components
...
-import { Text } from '../';
+import { ThemePreview, CustomPropsTable } from 'docs';
+import { Text } from '@deity/falcon-ui';
 
-# Text
+<ThemePreview of={Text} /> 
 
-### Basic usage
+`Text` component renders as `<p>` element by default and accepts [responsive, theme-based style props](/responsive-props)
+ that allow it's further styling customizations.
```
Comment (Borales) [inline packages/falcon-ui/src/components/Text.mdx]:
```suggestion
 that allow its further styling customizations.
```

File: packages/falcon-ui/src/index.mdx [hunk 3]
```diff
@@ -4,39 +4,53 @@ route: /
...
+## Features
 
-```sh
+- over **30 UI components** that are great basis for any component based design system
+
+- super small - **~17KB** gzipped, ~55KB minified - including all dependencies ([emotion 10](https://github.com/emotion-js/emotion))
+
+- **tree shaking friendly** - it really works even with [webpack code splitting](https://github.com/webpack/webpack/issues/7782)
+
+- build in [**design constraints theme**](/theming) that improves maintability and consistency of UI immensely
```
Comment (Borales) [inline packages/falcon-ui/src/index.mdx]:
```suggestion
- built-in [**design constraints theme**](/theming) that improves maintability and consistency of UI immensely
```

File: packages/falcon-ui/src/theme/responsiveprops.mdx [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+import { Playground } from 'docz';
+import { Box, GridLayout, Divider, Button } from '@deity/falcon-ui';
+import { ResponsivePropsTable, ViewThemeSection } from 'docs';
+
+
+# Responsive theme-based props
+
+## Introduction
+
+All **`@deity/falcon-ui`** components accept, responsive, theme-based style props which offer easy to use, flexible yet consistent styling customization.
```
Comment (Borales) [inline packages/falcon-ui/src/theme/responsiveprops.mdx]:
```suggestion
All **`@deity/falcon-ui`** components accept responsive, theme-based style props which offer easy to use, flexible yet consistent styling customization.
```

File: packages/falcon-ui/src/theme/responsiveprops.mdx [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+const theme = createTheme({
+  components: {
+    button: {
+      px: 'lg',
+      bg: 'primaryLight'
+    }
+  }
+});
+```
+All of `@deity/falcon-ui` components come with default theme which can be overriden as shown in example above.
```
Comment (Borales) [inline packages/falcon-ui/src/theme/responsiveprops.mdx]:
```suggestion
All of `@deity/falcon-ui` components come with default theme which can be overridden as shown in the example above.
```

File: packages/falcon-ui/src/theme/responsiveprops.mdx [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+
+## Responsive notation
+All of the responsive theme-based props work seamlessy with `breakpoints` defined in `Theme`.
+
+<ViewThemeSection section="breakpoints">
+  view available theme breakpoints
+</ViewThemeSection>
+
+Simply pass object as responsive prop value with it's keys being breakpoints keys defined in `Theme` and values being property values to have responsive prop that changes based on
```
Comment (Borales) [inline packages/falcon-ui/src/theme/responsiveprops.mdx]:
```suggestion
Simply pass an object as responsive prop value with its keys being breakpoints keys defined in `Theme` and values being property values to have a responsive prop that changes based on 
```

File: packages/falcon-ui/src/theme/theme.mdx [hunk 7]
```diff
@@ -4,18 +4,138 @@ route: /theming
...
+import { Playground } from 'docz';
+import { ViewThemeSection, ActivateComponentFinder, Card, customizedTheme, customizedButtonTheme } from 'docs';
+import { ThemeProvider, H1, H2, Button, FlexLayout } from '@deity/falcon-ui';
+
 # Theming
+
+
 ## Introduction
+**`@deity/falcon-ui`** ships with powerful `ThemeProvider` based theming capabilities.
+Every component has default theme build-in, providing consistent and pleasing look and feel out of the box, yet giving possibility to customize it in every detail.
```
Comment (Borales) [inline packages/falcon-ui/src/theme/theme.mdx]:
```suggestion
Every component has default theme built-in, providing a consistent and pleasing look and feel out of the box, yet giving a possibility to customize it in every detail.
```

File: packages/falcon-ui/src/theme/theme.mdx [hunk 8]
```diff
@@ -4,18 +4,138 @@ route: /theming
...
+  </ThemeProvider>
+);
+```
+
+### Additional resources
+- [Constructing design system](https://medium.freecodecamp.org/how-to-construct-a-design-system-864adbf2a117)
+- [8px grid system](https://builttoadapt.io/intro-to-the-8-point-grid-system-d2573cde8632)
+
+## Customizing theme
+All theme customizations are done using `createTheme` function by providing necessary overrides and passing it's output as `theme` prop to `ThemeProvider`.
```
Comment (Borales) [inline packages/falcon-ui/src/theme/theme.mdx]:
```suggestion
All theme-related customizations are done using `createTheme` function by providing necessary overrides and passing its output as `theme` prop to `ThemeProvider`.
```

File: packages/falcon-ui/src/theme/theme.mdx [hunk 9]
```diff
@@ -4,18 +4,138 @@ route: /theming
...
+    }
+  }
+});
+```
+Every component accepts all of the [responsive, theme-based style props](/responsive-props) as well as [`css`](/responsive-props#css-prop) prop.
+
+<ActivateComponentFinder />
+
+## Custom themed components
+  In addition to build-in components there is an easy way to create custom components with default theme that also plugin-in into general theming system.
```
Comment (Borales) [inline packages/falcon-ui/src/theme/theme.mdx]:
```suggestion
  In addition to built-in components, there is an easy way to create custom components with a default theme that also plugin-in into the general theming system.
```

File: packages/falcon-ecommerce-uikit/src/Footer/Newsletter.tsx [hunk 10]
```diff
@@ -26,11 +26,17 @@ export const Newsletter = () => (
...
+          css={({ theme }) => ({
+            height: theme.spacing.lg
+          })}
+        />
         <Button as="input" type="submit" value="Subscribe" />
       </Group>
-      <Label htmlFor="subscribe" mt="md">
-        <Checkbox id="subscribe" required size={16} mr="xs" />I would like to subscribe to updates
+      <Label htmlFor="subscribe" my="sm" display="flex" justifyContent="center" alignItems="center">
+        <Checkbox id="subscribe" required mr="xs" />I would like to subscribe to updates
```
[the remark this answers is not in the document]
Comment (tadeuszwojcik) [reply packages/falcon-ecommerce-uikit/src/Footer/Newsletter.tsx]: Definitely, but IMHO it should be separate issue - cleaning up ecommerce toolkit or similar

File: packages/falcon-ecommerce-uikit/src/MiniCart/MiniCart.tsx [hunk 11]
```diff
@@ -61,30 +61,28 @@ const MiniCartProducts: React.SFC<any> = ({ products }) => (
...
+  <ToggleMiniCartMutation>
+    {toggle => (
+      <React.Fragment>
+        <Sidebar as={Portal} visible={open} side="right">
+          <SidebarLayout>
+            <Icon src="close" onClick={() => toggle()} position="absolute" top={15} right={30} />
+            <H2 mb="lg">Shopping cart</H2>
+            <MiniCartProducts products={basketItems} />
+            <Box position="absolute" bottom={0} right={0} left={0} bg="secondaryLight" p="sm">
+              <Button css={{ width: '100%' }}>Checkout</Button>
```
[the remark this answers is not in the document]
Comment (tadeuszwojcik) [reply packages/falcon-ecommerce-uikit/src/MiniCart/MiniCart.tsx]: Same here

File: packages/falcon-theme-editor/package.json [hunk 12]
```diff
@@ -13,7 +13,7 @@
     "src"
   ],
   "peerDependencies": {
-    "@deity/falcon-ui": "0.0.1",
+    "@deity/falcon-ui": "0.0.2",
```
[the remark this answers is not in the document]
Comment (tadeuszwojcik) [reply packages/falcon-theme-editor/package.json]: I'm not sure about that, if it was stable version then yes, but right now I don't know.

[the remark this answers is not in the document]
Comment (Borales) [reply packages/falcon-theme-editor/package.json]: @tadeuszwojcik I mean that it's a peerDependency, we should be safe to set a bottom edge

File: packages/falcon-ecommerce-uikit/src/Category/Category.tsx [hunk 13]
```diff
@@ -1,4 +1,6 @@
 import React, { MouseEventHandler } from 'react';
+// eslint-disable-next-line
+import { NetworkStatus } from 'apollo-client';
```
[the remark this answers is not in the document]
Comment (tadeuszwojcik) [reply packages/falcon-ecommerce-uikit/src/Category/Category.tsx]: hmm, it seems to be used for ShowMore component in line 107

File: packages/falcon-ecommerce-uikit/src/Footer/Newsletter.tsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (tadeuszwojcik) [reply packages/falcon-ecommerce-uikit/src/Footer/Newsletter.tsx]: https://github.com/deity-io/falcon/issues/120

File: packages/falcon-ecommerce-uikit/src/MiniCart/MiniCart.tsx (hunk 11 above)
[the remark this answers is not in the document]
Comment (tadeuszwojcik) [reply packages/falcon-ecommerce-uikit/src/MiniCart/MiniCart.tsx]: https://github.com/deity-io/falcon/issues/120

File: packages/falcon-ui/src/components/Divider.mdx [hunk 14]
```diff
@@ -4,13 +4,17 @@ menu: Components
...
 
 import { Playground } from 'docz';
+import { ThemePreview } from 'docs';
+import { Divider } from '@deity/falcon-ui';
 
-import { Divider } from '../';
+<ThemePreview of={Divider} />
 
-# Divider
+`Divider` component renders by default as `<hr/>` and accepts all of the [responsive, theme-based style props](/responsive-props) that allow further styling customizations.
```
[the remark this answers is not in the document]
Comment (tadeuszwojcik) [reply packages/falcon-ui/src/components/Divider.mdx]: I'm not sure about this one as well, leaving it as is for now,  hopefully we'll get review from community anyway (native English speakers).

[the remark this answers is not in the document]
Comment (cane-lagerwaard) [reply packages/falcon-ui/src/components/Divider.mdx]: let's compromise: "component is rendered by default as"

**Pull request merged**